pkg/controller_helper/modelsource/modelhub.go

```diff
@@ -62,16 +62,20 @@ func (p *ModelHubProvider) ModelPath(skipModelLoader bool) string {
 	return CONTAINER_MODEL_PATH + "models--" + strings.ReplaceAll(p.modelID, "/", "--")
 }
 
-func (p *ModelHubProvider) InjectModelLoader(template *coreapplyv1.PodTemplateSpecApplyConfiguration, index int) {
+func (p *ModelHubProvider) InjectModelLoader(template *coreapplyv1.PodTemplateSpecApplyConfiguration, index int, initContainerImage string) {
 	initContainerName := MODEL_LOADER_CONTAINER_NAME
 	if index != 0 {
 		initContainerName += "-" + strconv.Itoa(index)
 	}
 
+	if initContainerImage == "" {
+		initContainerImage = pkg.LOADER_IMAGE
```

Comment:
I think, we can totally remove the LOADER_IMAGE in the follow up. The configmap is the single source of truth.

Reply:
Updated. I added a validation function to avoid an empty initContainerImage in the ConfigMap.